File: 1460-NumberOfSubstringsContainingAllThreeCharacters/1460-NumberOfSubstringsContainingAllThreeCharacters.java
// Last updated: 02/01/2026, 10:08:20
class Solution {
    public int numberOfSubstrings(String s) {
        int arr[]=new int[3];
        for(int i=0;i<3;i++){
            arr[i]=-1;
        }
        int count=0;
        for(int i=0;i<s.length();i++){
            arr[s.charAt(i)-'a']=i;
            if(arr[0]!=-1&&arr[1]!=-1&&arr[2]!=-1){
                count=count+(Math.min(arr[2],Math.min(arr[0],arr[1])))+1;
            }
        }
        return count;
    }
}

File: 1460-NumberOfSubstringsContainingAllThreeCharacters/1460-NumberOfSubstringsContainingAllThreeCharacters.kt
// Last updated: 02/01/2026, 10:08:23
class Solution {
    fun numberOfSubstrings(s: String): Int {
        val freq = IntArray(3)
        var count = 0
        var curCount = 0
        var left = 0
        val n = s.length

        for (right in 0 until n) {
            val ch = s[right]
            freq[ch - 'a']++
            if (freq[ch - 'a'] == 1) {
                curCount++
            }
            while (curCount == 3) {
                count += (n - right)
                val c = s[left]
                freq[c - 'a']--
                if (freq[c - 'a'] == 0) {
                    curCount--
                }
                left++
            }
        }
        return count
    }
}
